lib/OntologyAPI/utils/misc.py

```diff
@@ -0,0 +1,31 @@
+import re
+import time
+import jsonschema
+
+from OntologyAPI.utils import config, schema, namespace
+from OntologyAPI.exceptions import InvalidParamsError, REError
+from jsonschema.exceptions import ValidationError
+
+_CONF=config.get_config()
+_SCHEMAS=schema.load_schemas()
+_NAMESPACE=namespace.load_namespaces()
+
+def validate_params(params, schema='default'):
+    _params={}
+    try:
+        _params=params.copy()
+        _params['ts'] = _params.get('ts', int(time.time() * 1000)) 
+        _params['ns'] = _params.get('ns', 'default') 
+        _ns = _NAMESPACE[_params['ns']]
+        _schema = _SCHEMAS[schema]
+        for x in ['@onto_terms', '@onto_edges']:
```

Comment:
Rather than hard coding these field names here and writing a loop, it might be better to assign defaults with:
```py
_params.update(_ns['query_defaults'])
```

Reply:
some are not just simply taking the default value. The schema will be taken into account when assign values. So I will leave it as is